### PR TITLE
Deprecate Bbox.anchored() with no container.

### PR DESCRIPTION
**Pull request by anntzer**
This previously returned a copy of the bbox itself, and was completely unused; let's deprecate this for clarity ("anchoring a bbox to itself" doesn't make much sense anyways).

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: doc/api/next_api_changes/deprecations/24913-AL.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``bbox.anchored()`` with no explicit container
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If no *container* is passed to `.BboxBase.anchored`, the call simply returns a
+frozen copy of the original bbox; this is now deprecated.
```
Comment (jklymak) [inline doc/api/next_api_changes/deprecations/24913-AL.rst:4]:
```suggestion
Not passing a *container* argument to `.BboxBase.anchored` is now deprecated.
```

**Review by jklymak — APPROVED** [review]
You can take or leave my suggested change - I found it hard to parse the sentence, and then was told it was deprecated anyway - seems easier to just deprecate ;-)

File: doc/api/next_api_changes/deprecations/24913-AL.rst (hunk 1 above)
Comment (anntzer) [reply doc/api/next_api_changes/deprecations/24913-AL.rst:4]: done

Comment (jklymak) [reply doc/api/next_api_changes/deprecations/24913-AL.rst]: Anyone should merge when green enough....
